src/mainchain/MainchainGatewayV3.sol

```diff
@@ -58,81 +61,16 @@ contract MainchainGatewayV3 is
     _disableInitializers();
   }
 
-  fallback() external payable {
```

Comment:
Any requirement for this? Otherwise should not remove.

Reply:
an attempt to reduce code size, we also do not need this function too.

Reply:
I think it's not necessary (just a few bytes) and might break things. Please do not remove it.

Reply:
If this function is kept, the current restrict logic can be bypassed using this function to create a deposit when deposit is restricted.